### PR TITLE
Add ParallelInfo executable and build system info

### PR DESCRIPTION
**Pull request by nilsdeppe**

### CODE REVIEW

File: cmake/SetOutputDirectory.cmake [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
```
Comment (kidder) [inline cmake/SetOutputDirectory.cmake:1]: add license info

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+.ipynb_checkpoints
+
+# charm++ runtime files
+charmrun.*
+
+# Observer output
+*.xmf
+*.h5
+
+# compile commands used by tools like clang-tidy, cppcheck
```
Comment (nilsdeppe) [inline .gitignore]: remove

File: cmake/SetOutputDirectory.cmake [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+# Set the directory where the library and executables are placed
+# The default can be overridden by specifying
+# `-D CMAKE_RUNTIME_OUTPUT_DIRECTORY=/path/`
+if (NOT CMAKE_RUNTIME_OUTPUT_DIRECTORY)
+    set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin/")
```
Comment (nilsdeppe) [inline cmake/SetOutputDirectory.cmake]: Add help string  to cache

File: src/Informer/InfoFromBuild.cpp [hunk 4]
```diff
@@ -0,0 +1,45 @@
+
+#include "Informer/InfoFromBuild.hpp"
+
+#include <sstream>
+
+#ifndef STRINGIZE
+#define STRINGIZE(x) #x
```
Comment (nilsdeppe) [inline src/Informer/InfoFromBuild.cpp]: use `BOOST_PP_STRINGIZE`

File: src/Informer/InfoFromBuild.cpp [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+int spectre_major_version() { return @SpECTRE_VERSION_MAJOR@; }
+
+int spectre_minor_version() { return @SpECTRE_VERSION_MINOR@; }
+
+int spectre_patch_version() { return @SpECTRE_VERSION_PATCH@; }
+
+std::string info_from_build() {
+  static const std::string info = [] {
+    std::ostringstream os;
+    os << "\nSpECTRE Build Information:\n";
```
Comment (nilsdeppe) [inline src/Informer/InfoFromBuild.cpp]: remove newline at beginning

**Review by kidder — APPROVED** [review]

**Pull request merged**